notifications_api_common/management/commands/register_kanalen.py

```diff
@@ -26,7 +50,15 @@ def create_kanaal(kanaal: str) -> None:
     # look up the exchange in the registry
     _kanaal = next(k for k in KANAAL_REGISTRY if k.label == kanaal)
 
-    kanalen = client.get("kanaal", params={"naam": kanaal})
+    response_data = []
+
+    try:
+        response: Response = client.get("kanaal", params={"naam": kanaal})
+        kanalen: list[dict] = response.json() or []
+        response.raise_for_status()
+    except (RequestException, JSONDecodeError) as exception:
+        raise KanaalRequestException(kanaal=kanaal, data=response_data) from exception
+
     if kanalen:
         raise KanaalExists()
```

Comment:
I think it would be nice for this exception also to inherit from `KanaalException`